src/components/webview/MainProvider.ts

```diff
@@ -29,10 +30,40 @@ import {
 	CodemodNodeHashDigest,
 	selectCodemodArguments,
 } from '../../selectors/selectCodemodTree';
-import { isNeitherNullNorUndefined } from '../../utilities';
+import { buildTypeCodec, isNeitherNullNorUndefined } from '../../utilities';
 
 const X_INTUITA_ACCESS_TOKEN = 'X-Intuita-Access-Token'.toLocaleLowerCase();
 
+export const validateAccessToken = async (
+	accessToken: string,
+): Promise<boolean> => {
+	const result = await axios.post(
+		'https://telemetry.intuita.io/validateAccessToken',
+		{},
+		{
+			headers: {
+				[X_INTUITA_ACCESS_TOKEN]: accessToken,
+			},
+		},
+	);
+	const { data, status } = result;
+	if (status !== 200) {
```

Comment:
I think now you don't need that, in general, if the promise rejects, you are not logged in (you may inspect the error code for 401 or other code to know what exactly the problem is).

Reply:
addressed